### PR TITLE
Change fetching to use swr library

### LINKED ISSUE #16: Fetch errors/users/etc. is broken
I think #13 caused collection fetching on https://otp-admin-ui.ibi-group.now.sh/ to break.

I believe the `componentDidUpdate` was needed for the vercel/now-hosted version of the UI to work.

Related: might be time to implement #10.

### PR DESCRIPTION
**Pull request by landonreed**
This is a somewhat big refactor, but simplifies things quite a bit. Fixes #16 and #10. A lot of lines changed, but part of this is just re-indentation due to the change from class to functional components.

One notable change is that the fetch button was removed, but since SWR kind of more or less handles auto-refetching and stabilizes the fetched data, I don't think that's a big loss.

https://swr.vercel.app/docs/with-nextjs#client-side-data-fetching

### CODE REVIEW

File: components/ErrorEventsDashboard.js [hunk 1]
```diff
@@ -1,134 +1,90 @@
...
-    const { events, eventsError, fetchMessage } = this.state
-    if (!auth.isAuthenticated) return null
-    const hasEvents = events && events.length > 0
-    return (
-      <div>
-        <h2>Error Events Summary</h2>
-        <div className='controls'>
-          <Button onClick={this.handleFetchErrors}>
-            Fetch errors
-          </Button>
```
Comment (evansiroky) [inline components/ErrorEventsDashboard.js:56]: This button to refetch is gone. Granted, with SWR, it gets automatically refetched every 30 seconds, but would it be useful to have a way to refetch immediately or at least show that refetching will occur in x# of seconds?

File: components/RequestLogsDashboard.js [hunk 2]
```diff
@@ -1,113 +1,68 @@
...
-    const { auth, isAdmin } = this.props
-    const { logs, logsError, fetchMessage } = this.state
-    if (!auth.isAuthenticated) return null
-    return (
-      <div>
-        <h2>Request Log Summary</h2>
-        <div className='controls'>
-          <Button onClick={this.handleFetchLogs}>
-            Fetch logs
-          </Button>
```
Comment (evansiroky) [inline components/RequestLogsDashboard.js:56]: This button to refetch is gone. Granted, with SWR, it gets automatically refetched every 30 seconds, but would it be useful to have a way to refetch immediately or at least show that refetching will occur in x# of seconds?

**Review by evansiroky — CHANGES_REQUESTED** [review]
> One notable change is that the fetch button was removed, but since SWR kind of more or less handles auto-refetching and stabilizes the fetched data, I don't think that's a big loss.

There is something lost in that it isn't immediately clear that the data isn't automatically refreshed, so as noted in comments, I think it'd be good to either have a way to refresh immediately or to note the auto-refreshing.

File: components/ErrorEventsDashboard.js (hunk 1 above)
Comment (landonreed) [reply components/ErrorEventsDashboard.js:56]: Resolved in https://github.com/ibi-group/otp-admin-ui/pull/17/commits/7c774280b1d29ef63708133ecd38062515fb33ee (added auto fetch message and button, which I think makes for a nice approach).

**Review by evansiroky — APPROVED** [review]
Good to go after making the Travis build pass

File: components/UserList.js [hunk 3]
```diff
@@ -55,110 +37,102 @@ class UserList extends Component {
...
-    const {auth, type} = this.props
-    const { accessToken } = auth
-    const email = window.prompt(`Enter an email address for ${type} user.`)
+  const onCreateAdminUser = async () => {
+    const email = window.prompt(`Enter an email address for admin user.`)
     // TODO: Validate user.
     if (!email) return
     // Create user and re-fetch users.
+    const adminUrl = _getUrl('admin')
+    // TODO: Can we replace with useSWR (might only be possible for fetching/GET)?
```
Comment (binh-dam-ibigroup) [inline components/UserList.js:57]: Isn't mutate the way SWR does that?

**Review by binh-dam-ibigroup — APPROVED** [review]
Much much cleaner this way. Very nice.

File: components/UserList.js (hunk 3 above)
Comment (landonreed) [reply components/UserList.js:57]: No, this TODO comment is from earlier in my research into SWR. SWR is basically only for fetching. `mutate` is used to indicate that the data fetched needs to be re-fetched, not actually to mutate the data. I was confused by this naming also.

**Pull request merged**